### PR TITLE
Add a caption_entity filter for filtering caption entities

### PR DESCRIPTION
**Pull request by PaulSonOfLars**
Currently, the Filters.entity filter only checks message.entities.
This PR adds Filters.caption_entities to check message.caption_entities.

### CODE REVIEW

**Comment by keessonnema** [comment]
This would be really helpful. Code looks good aswell. +1

File: telegram/ext/filters.py [hunk 1]
```diff
@@ -534,6 +534,27 @@ def __init__(self, entity_type):
...
+                in :class:`telegram.MessageEntity`.
+
+        """
+
+        def __init__(self, entity_type):
+            self.entity_type = entity_type
+            self.name = 'Filters.caption_entity({})'.format(self.entity_type)
+
+        def filter(self, message):
+            return any([entity.type == self.entity_type for entity in message.caption_entities])
```
Comment (jh0ker) [inline telegram/ext/filters.py]: nitpick, but why not just use a generator expression instead of a list comprehension?

Comment (PaulSonOfLars) [reply telegram/ext/filters.py]: totally agree, i copy pasted the previous filter code is why

**Comment by jh0ker** [comment]
Thanks for your contribution :slightly_smiling_face:

**Pull request merged**